ceno_emul/src/platform.rs

```diff
@@ -60,21 +69,21 @@ impl Platform {
     // Startup.
 
     pub const fn pc_base(&self) -> Addr {
-        self.rom.start
+        self.prog_code.start
     }
 
     // Permissions.
 
     pub fn can_read(&self, addr: Addr) -> bool {
-        self.is_rom(addr) || self.is_ram(addr) || self.is_pub_io(addr) || self.is_hints(addr)
+        self.can_write(addr)
```

Comment:
This completely precludes read-only memory?

Reply:
Yes, there are no read-only memory anywhere.